File: westcoast/src/main/java/frc/commands/Shoot.java
package frc.commands;

import edu.wpi.first.wpilibj2.command.CommandBase;

public class Shoot extends CommandBase {
  private boolean end;

  public Shoot() {
  }

  @Override
  public void initialize() {    
    System.out.println("shooting");
  }

  @Override
  public void execute() {
    end = true;
  }

  @Override
  public void end(boolean interrupted) {
  }

  @Override
  public boolean isFinished() {
    return end;
  }
}

File: westcoast/src/main/kotlin/frc/commands/magazine/Shoot.kt
package frc.commands.magazine

import edu.wpi.first.wpilibj2.command.CommandBase
import frc.subsystems.MagazineSubsystem

class Shoot(
    private val magazine: MagazineSubsystem
) : CommandBase() {
  override fun initialize() {
    magazine.shooterTune()
  }

  override fun execute() {
    magazine.runShooter(0.78)
    magazine.runMagMotor(0.3)
    magazine.runIndexMotor(0.25)
  }

  override fun end(interrupted: Boolean) {
    magazine.stopMotors()
    magazine.resetIndexCount()
  }

  init {
    addRequirements(magazine)
  }
}